### ISSUE TITLE
Calling `hardhat_setStorageAt` after `eth_getStorageAt` doesn't work when forking

### ISSUE THREAD

**Issue by Rubilmax** [NomicFoundation/edr#503]
# Minimal reproducible example

Using hardhat 2.22.5 ([EDR ^0.4.0](https://github.com/NomicFoundation/hardhat/blob/bdd04d37f2ee41f68943f60bf0fd3ac5546a05ad/packages/hardhat-core/package.json#L105)), the following code rejects with "Contract with code hash `0xaee43571f51ad4922c41ab126fb98a76934d579c76f54b9e393ade09aad21646` does not exist.":

```typescript
    await getStorageAt(
      "0xbe9895146f7af43049ca1c1ae358b0541ea49704",
      "0x4f039c94bc7b6c8e7867b9fbd2890a637837fea1c829f434a649c572b15b2969",
    );

    await setStorageAt(
      "0xbe9895146f7af43049ca1c1ae358b0541ea49704",
      "0x4f039c94bc7b6c8e7867b9fbd2890a637837fea1c829f434a649c572b15b2969",
      "0x0000000000000000000000000000000000000000000000000000000000000001",
    );
```

While the following one does not:

```typescript
    await setStorageAt(
      "0xbe9895146f7af43049ca1c1ae358b0541ea49704",
      "0x4f039c94bc7b6c8e7867b9fbd2890a637837fea1c829f434a649c572b15b2969",
      "0x0000000000000000000000000000000000000000000000000000000000000001",
    );

    await getStorageAt(
      "0xbe9895146f7af43049ca1c1ae358b0541ea49704",
      "0x4f039c94bc7b6c8e7867b9fbd2890a637837fea1c829f434a649c572b15b2969",
    );
```

In other words, inverting the order of requests `eth_getStorageAt` and `hardhat_setStorageAt` changes the behavior of the EDR, while it should not (both should not reject)

**Comment by fvictorio** [comment]
I can't reproduce this. @Rubilmax could you please create and share a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Comment by Rubilmax** [comment]
Sorry, update the description because I mixed up addresses.

Here is a reproducible example: https://github.com/Rubilmax/edr-setStorageAt-issue

Configure a rpc URL in the environment variable and run `yarn test`

**Comment by Rubilmax** [comment]
Quick notes: [0xbe9895146f7af43049ca1c1ae358b0541ea49704](https://etherscan.io/address/0xbe9895146f7af43049ca1c1ae358b0541ea49704) corresponds to cbETH on mainnet

**Comment by Rubilmax** [comment]
Thanks a lot for the hot fix, will do this while a proper fix is published to the EDR

**Comment by Wodann** [comment]
I discovered the culprit and [fixed ](https://github.com/NomicFoundation/edr/pull/517) the bug. Thank you for the detailed reproduction steps!

We'll release a new patch version of EDR as soon as possible.

**Comment by fvictorio** [comment]
We've released [EDR v0.4.1](https://github.com/NomicFoundation/edr/releases/tag/%40nomicfoundation%2Fedr%400.4.1) with a fix for this issue. 

To upgrade to the latest EDR version immediately, you can remove your `node_modules` directory and npm/yarn/pnpm lock file and reinstall your dependencies. Otherwise you can wait for the next Hardhat release which will automatically use the latest EDR version.

You can verify that you’re using the latest version of EDR by running the following command: `npm ls @nomicfoundation/edr`.
